aiida_quantumespresso/cli/calculations/pw2wannier90.py

```diff
@@ -16,56 +18,75 @@
 from ..utils import options as options_qe
 from . import cmd_launch
 
-PARENT_FOLDER = OverridableOption(
-    '-F',
-    '--parent_folder',
-    'parent_folder',
-    metavar='RemoteData/FolderData',  # placeholder in the help text
-    type=types.NodeParamType(sub_classes=('aiida.data:remote', 'aiida.data:folder')),
-    help='A RemoteData or FolderData node identified by its ID or UUID.'
-)
-
-SINGLE_FILE = OverridableOption(
+NNKP_FILE = OverridableOption(
     '-S',
-    '--single_file',
-    'single_file',
-    metavar='SinglefileData',  # placeholder in the help text
+    '--nnkp-file',
     type=types.NodeParamType(sub_classes=('aiida.data:singlefile',)),
-    help='A SinglefileData node identified by its ID or UUID.'
+    required=True,
+    help='SinglefileData containing the .nnkp file generated by a wannier90.x preprocessing.'
+)
+
+SCDM_MODE = OverridableOption(
+    '--scdm-mode',
+    type=click.Choice(['no', 'isolated', 'erfc', 'gaussian']),
+    default='no',
+    show_default=True,
+    help='Whether to use the SCDM algorithm to determine the UNK matrices.'
+)
+
+WRITE_UNK = OverridableOption(
+    '-u',
+    '--write-unk',
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help=u'Output also the UNK matrices (for real-space plotting of the Wannier functions).'
 )
 
 
 @cmd_launch.command('pw2wannier90')
 @options.CODE(required=True, type=types.CodeParamType(entry_point='quantumespresso.pw2wannier90'))
-@PARENT_FOLDER(required=True, help='RemoteData folder containing the output of a PW calculation')
-@SINGLE_FILE(required=True, help='SinglefileData containing the .nnkp file generated by a wannier90.x preprocessing')
+@options_qe.PARENT_FOLDER(required=True, help='RemoteData node containing the output of a PW NSCF calculation.')
+@NNKP_FILE()
+@SCDM_MODE()
+@WRITE_UNK()
 @options_qe.MAX_NUM_MACHINES()
 @options_qe.MAX_WALLCLOCK_SECONDS()
 @options_qe.WITH_MPI()
 @options_qe.DAEMON()
 @decorators.with_dbenv()
-def launch_calculation(code, parent_folder, single_file, max_num_machines, max_wallclock_seconds, with_mpi, daemon):
+def launch_calculation(
+    code, parent_folder, nnkp_file, scdm_mode, write_unk, max_num_machines, max_wallclock_seconds, with_mpi, daemon
+):
     """Run a Pw2wannier90Calculation with some sample parameters and the provided inputs."""
     from aiida.orm import Dict
     from aiida.plugins import CalculationFactory
     from aiida_quantumespresso.utils.resources import get_default_options
 
     parameters = {
-        'inputpp': {
+        'INPUTPP': {
             'write_amn': True,
             'write_mmn': True,
-            'write_unk': False,
-            'scdm_proj': True,
-            'scdm_entanglement': 'isolated',
         }
     }
 
+    if write_unk:
+        parameters['INPUTPP']['write_unk'] = True
+    else:
+        parameters['INPUTPP']['write_unk'] = False
+
+    if scdm_mode != 'no':
```

Comment:
it doesn't seem that the actual value of `scdm_mode` gets used so why does the option specify those other possible values `'isolated', 'erfc', 'gaussian'`?

Reply:
You are right! I've fixed it.